Gems/Atom/RPI/Code/Include/Atom/RPI.Edit/Material/MaterialPipelineSourceData.h

```diff
@@ -39,16 +40,22 @@ namespace AZ
 
                 AZStd::string m_shader; //!< Relative path to a template .shader file that will configure the final shader asset.
                 AZStd::string m_azsli; //!< Relative path to a template .azsli file that will be stitched together with material-specific shader code.
+
                 Name m_shaderTag; //!< tag to identify the shader, particularly in lua functors
+                //! Type of the generated draw-items
+                using DrawItemType = AZ::RPI::ShaderCollection::Item::DrawItemType;
+                DrawItemType m_drawItemType = DrawItemType::Raster;
 
                 bool operator==(const ShaderTemplate& rhs) const
                 {
-                    return m_shader == rhs.m_shader && m_azsli == rhs.m_azsli && m_shaderTag == rhs.m_shaderTag;
+                    return m_shader == rhs.m_shader && m_azsli == rhs.m_azsli && m_shaderTag == rhs.m_shaderTag &&
+                        m_drawItemType == rhs.m_drawItemType;
                 }
 
                 bool operator<(const ShaderTemplate& rhs) const
                 {
-                    return AZStd::tuple(m_shader, m_azsli, m_shaderTag.GetHash()) < AZStd::tuple(rhs.m_shader, rhs.m_azsli, rhs.m_shaderTag.GetHash());
+                    return AZStd::tuple(m_shader, m_azsli, m_shaderTag.GetHash(), static_cast<int32_t>(m_drawItemType)) <
+                        AZStd::tuple(rhs.m_shader, rhs.m_azsli, rhs.m_shaderTag.GetHash(), static_cast<int32_t>(rhs.m_drawItemType));
```

Comment:
Consider using `AZStd::tie` instead of `AZStd::tuple` here. The tuple constructor will copy the strings, while `tie` will not.

```suggestion
                    return AZStd::tie(m_shader, m_azsli, m_shaderTag.GetHash(), static_cast<int32_t>(m_drawItemType)) <
                        AZStd::tie(rhs.m_shader, rhs.m_azsli, rhs.m_shaderTag.GetHash(), static_cast<int32_t>(rhs.m_drawItemType));
```

Reply:
AZStd::tie doesn't like R-values. I've changed the strings to string-view there to avoid copying them.